public/app/plugins/datasource/loki/syntax.ts

```diff
@@ -49,7 +49,67 @@ const AGGREGATION_OPERATORS: CompletionItem[] = [
   },
 ];
 
+export const PIPE_PARSERS: CompletionItem[] = [
+  {
+    label: 'json',
+    insertText: 'json',
+    documentation: 'Extracting labels from the log line using json parser. Only available in Loki 2.0+.',
+  },
+  {
+    label: 'regexp',
+    insertText: 'regexp',
+    documentation: 'Extracting labels from the log line using regexp parser. Only available in Loki 2.0+.',
+  },
+  {
+    label: 'logfmt',
+    insertText: 'logfmt',
+    documentation: 'Extracting labels from the log line using logfmt parser. Only available in Loki 2.0+.',
+  },
+];
+
+export const PIPE_OPERATORS: CompletionItem[] = [
+  {
+    label: 'label_filter',
+    insertText: 'label_filter',
+    documentation: 'Only available in Loki 2.0+.',
+  },
+  {
+    label: 'label_format',
+    insertText: 'label_format',
+    documentation: 'Only available in Loki 2.0+.',
+  },
+  {
+    label: 'line_format',
+    insertText: 'line_format',
+    documentation: 'Only available in Loki 2.0+.',
+  },
+];
+
 export const RANGE_VEC_FUNCTIONS = [
+  {
+    insertText: 'avg_over_time',
+    label: 'avg_over_time',
+    detail: 'avg_over_time(range-vector)',
+    documentation: 'The average of all values in the specified interval.',
+  },
+  {
+    insertText: 'min_over_time',
+    label: 'min_over_time',
+    detail: 'min_over_time(range-vector)',
+    documentation: 'The minimum of all values in the specified interval.',
+  },
+  {
+    insertText: 'max_over_time',
+    label: 'max_over_time',
+    detail: 'max_over_time(range-vector)',
+    documentation: 'The maximum of all values in the specified interval.',
+  },
+  {
+    insertText: 'sum_over_time',
```

Comment:
We also have those:

`stdvar_over_time(range vec)` , `stddev_over_time(range vec)` and `quantile_over_time(value between 0 and 1,range vec)`

In the doc I was saying we won't do it but I just did them anyway as I think they are killer feature.

Reply:
Full list here: https://github.com/grafana/loki/compare/master...cyriltovena:logql-parser#diff-8c1350fd75c7775d87e4408faa8c5fc2R293

Reply:
I have added those. One question, should we add also `BYTES_OVER_TIME` and `BYTES_RATE`?

Reply:
yes we should.

BYTES_OVER_TIME => gives you the bytes count of logs over time
BYTES_RATE => gives you the bytes per second of logs over time

Reply:
Thanks! Not sure why they weren't there before, but I have added them as well. 🙂 Also regarding this: 

>In the doc I was saying we won't do it but I just did them anyway as I think they are killer feature.

🔥🔥🔥🔥🔥